src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.command.*;
+import duke.models.Deadline;
+import duke.models.Event;
+import duke.models.Task;
+import duke.models.Todo;
+import duke.ui.Ui;
+
+
+/**
+ * This class returns a command based on the user input
+ */
+public class Parser {
+    private Ui ui;
+    private TaskList taskList;
+
+    /**
+     * Initialise a Parser object
+     * @param ui
+     * @param tasks
+     */
+    public Parser(Ui ui, TaskList tasks) {
+        this.ui = ui;
+        this.taskList = tasks;
+    }
+
+    /**
+     * Requirement: deals with making sense of the user command
+     *
+     * Takes the input given by the user, extracts the verb
+     * and translates it into a command
+     *
+     * @param command
+     * @return
+     */
+    public Command parse(String command) {
+        // Idea from : https://stackoverflow.com/questions/70683058/using-startswith-in-switch-case-in-java
```

Comment:
I like that you included comments. The comments helped me to understand the code better